src/main/scala/firrtl/transforms/TopWiring.scala

```diff
@@ -0,0 +1,269 @@
+            case Some(xs:Seq[(ComponentName, Type, Boolean, InstPath, String)]) =>
+              sourceMap.update(currentmodule.name, xs :+
+                 (ComponentName(w.name,currentmodule), tpe, isport ,Seq[String](w.name), prefix))
+            case None =>
+              sourceMap(currentmodule.name) = Seq((ComponentName(w.name,currentmodule),
+                                                   tpe, isport ,Seq[String](w.name), prefix))
+          }
+      }
+      w // Return argument unchanged (ok because DefWire has no Statement children)
+    // If not, apply to all children Statement
+    case _ => s map getSourceTypes(sourceList, sourceMap, currentmodule, state)
+  }
+
+
+
+  /** Get the Type of each port to be connected
+    *
+    * Similar to getSourceTypes, but specifically for ports since they are not found in statements.
+    * Find the definition of each port in sourceList, and get the type and whether or not it's a port
+    * Update the results in sourceMap
+    */
+  private def getSourceTypesPorts(sourceList: Map[ComponentName, String], sourceMap: mutable.Map[String,
+                          Seq[(ComponentName, Type, Boolean, InstPath, String)]],
+                          currentmodule: ModuleName, state: CircuitState)(s: Port): CircuitState = s match {
+    // If target port, add name and size to to sourceMap
+    case w: IsDeclaration =>
+      if (sourceList.keys.toSeq.contains(ComponentName(w.name, currentmodule))) {
+          val (isport, tpe, prefix) = w match {
+            case d: Port => (true, d.tpe, sourceList(ComponentName(w.name,currentmodule)))
+            case _ => throw new Exception(s"Cannot wire this type of declaration! ${w.serialize}")
+          }
+          val name = w.name
+          sourceMap.get(currentmodule.name) match {
+            case Some(xs:Seq[(ComponentName, Type, Boolean, InstPath, String)]) =>
+                sourceMap.update(currentmodule.name, xs :+
+                  (ComponentName(w.name,currentmodule), tpe, isport ,Seq[String](w.name), prefix))
+            case None =>
+                sourceMap(currentmodule.name) = Seq((ComponentName(w.name,currentmodule),
+                                                     tpe, isport ,Seq[String](w.name), prefix))
+          }
+      }
+      state // Return argument unchanged (ok because DefWire has no Statement children)
+    // If not, apply to all children Statement
+    case _ => state
+  }
+
+
+  /** Create a map of Module name to target wires under this module
+    *
+    * These paths are relative but cross module (they refer down through instance hierarchy)
+    */
+  private def getSourcesMap(state: CircuitState): Map[String,Seq[(ComponentName, Type, Boolean, InstPath, String)]] = {
+    val sSourcesModNames = getSourceModNames(state)
+    val sSourcesNames = getSourceNames(state)
+    val instGraph = new firrtl.analyses.InstanceGraph(state.circuit)
+    val cMap = instGraph.getChildrenInstances.map{ case (m, wdis) =>
```

Comment:
`getChildrenInstances` returns a `scala.collection.Map[String,mutable.Set[WDefInstance]]`.
Below, in line 148 `val seqChildren: Seq[(ComponentName,Type,Boolean,InstPath,String)] = cMap(mod.name).flatMap {`
values from `cMap` are iterated on and used to generate Seqs that are returned for use later. This *could* result in nondeterminism